src/tree.c

```diff
@@ -135,7 +135,7 @@ static git_tree_entry *alloc_entry(const char *filename)
 
 struct tree_key_search {
 	const char *filename;
-	uint16_t filename_len;
+	size_t filename_len;
```

Comment:
We only just moved to 16 bits. This is in part to reduce the size and improve the alignment of the tree entries, and partly to make sure we can allocate the entries in the pool. We figured that 16 bits is more than actual OSs allow you to use for file names, and we're not going to go out of our way to support people who should be using actual databases.